src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx

```diff
@@ -77,23 +102,31 @@ export default function Analysis() {
           <AnalysisSteps />
         ) : null}
         <Tabs
-          defaultActiveKey={
-            analysisContext.isError ||
-            (analysisTypesWithAdditionalPage.indexOf(
-              analysisContext.analysisType.type
-            ) > -1 &&
-              analysisContext.isCompleted)
-              ? analysisContext.analysisType.type
-              : "SETTINGS"
+          activeKey={
+            defaultTabKey === ""
+              ? analysisContext.isError
+                ? "job-error"
+                : analysisTypesWithAdditionalPage.indexOf(
+                    analysisContext.analysisType.type
+                  ) > -1 && analysisContext.isCompleted
+                ? analysisContext.analysisType.type.toLowerCase()
+                : "settings"
+              : defaultTabKey === "details" ||
+                defaultTabKey === "samples" ||
+                defaultTabKey === "share" ||
+                defaultTabKey === "delete"
+              ? "settings"
+              : defaultTabKey
```

Comment:
This is really complex, move it into a function.  If it does not need to be calculated on every re-render.

Reply:
Sure will do

Reply:
Fixed in 1f804bc95ac92e9c8447a801bd273330119acf44